src/docs/arch-column-row.dox

```diff
@@ -0,0 +1,99 @@
+column. WiredTiger supports both and each management system is represented internally by a certain
+type of B-Tree data structure.
+
+Keys and values in row-stores can be of variable length, thus, row-stores implement a generalized
+version of a B-Tree. Furthermore, keys are explicitly stored. Column-stores use a 64-bit unsigned
+integer record id as keys, thus, column-stores implement a specialized version of a B-Tree where the
+key has a predictable length. In the case of column-stores, keys are implied given the row's
+position in the B-Tree.
+
+There are two types of column-stores in WiredTiger:
+- The fixed length column-store, as its name implies, the value has a fixed length, and furthermore
+the value is restricted to between 1 and 8 bits in length. The bit length is specified when the
+column store is created. The fixed length column store has specialized use cases like bitmaps.
+- The variable length is a more general case which allows for values to have any length similarly to
+row-stores.
+
+Please refer to @ref arch-data-file to learn more about the on-disk format for row-stores and
+column-stores.
+
+@section column_row_rs Row-stores
+
+Row-stores are probably seen as the traditional access method and they serve a general purpose. In
+WiredTiger, row-store keys are explicitly stored and they can be duplicated within the same single
+file which directly impacts the on-disk storage size. However, keys are only duplicated if they are
+the first key on the page, which gets them copied/promoted to an internal page. They are not
+necessarily duplicated, too, they can be prefix/suffix compressed to discard unnecessary bytes on
+the internal pages.
+
+@section column_row_vlcs Variable length column-stores
+
+The main advantage of variable length column-stores is on-disk storage size. Their keys are not
+stored but derived from the row's position in the B-Tree. In fact, there is a starting key on each
+page which is used to figure out the rest of the keys present on that same page. Column-stores
+values also present an advantage to further save more disk space as they can be written in more
+compact forms through encoding techniques. In WiredTiger, the run-length encoding is used to replace
+any sequence of the same value with a count and value indicator. However, this makes the access to a
+specific record more complicated, it makes it impossible to jump to a record in a leaf page.
+
+@section column_row_flcs Fixed length column-stores
+
+Fixed length column stores are very different internally and usually serve specific purposes as it
+is a <a href="https://en.wikipedia.org/wiki/Bitmap"><b>Bitmap</b></a>. It makes fixed length
+column-stores efficient to retrieve a value at a given offset, the use of
+<a href="https://en.wikipedia.org/wiki/Bloom_filter"><b>Bloom filter</b></a> is probably one of the
+best examples.
+
+@section column_row_performance Performance
+
```

Comment:
See the discussion of performance above.  I'm afraid we'll need to remove this from the page, and perhaps move it to the regular part of the doc.  I'm sorry I wasn't clear about the scope of this page before this.

Reply:
No worries at all ! See 01b868718ef32d8c203f6c4cb6b5ddc1e8f3dcb0